azure/appservice-windows/main.tf

```diff
@@ -14,7 +14,8 @@ resource "azurerm_app_service_plan" "app" {
 }
 
 resource "azurerm_app_service" "app" {
-  name                = "as-${var.project}-${var.environment}"
+  count               = length(var.type)
```

Comment:
you need to use the `for_each` block instead, because the `count` interacts on a ordered list which can lead to unpleasant situations in the future (if you add a new app in the middle of an existent list of apps all the forward ones will be destroyed and recreated).

Reply:
I tried the for_each but wasn't working because it's a list, I even tried the toset() to use it with the for_each but no luck. I'll give it another go at it

Reply:
From the service template, we just need to pass a variable `apps` like below

```diff
module "whatever" {
  source              = "git::github.com/Nmbrs/terraform-modules//azure/appservice-windows?ref=main"    
  project             = var.project
  resource_group_name = var.resource_group
  environment         = var.environment
 apps       = var.apps
}
```


```
variable "apps" {
  default = {
    app1 = {
      key   = "app1"
      name  = "payroll"
      ...
    }

    app2 = {
      key   = "app2"
      name  = "identiy"
      ...
    }

    app3 = {
      key   = "app3"
      name  = "tentant"
      ...
    }
  }
}
```

Reply:
@pmmluis the problem with the count is that works well on ordered list, for example:

`list = ["apples","bananas","peaches"]`

then you apply (3 create, 0 changes, 0 destroy) and all goes well. Later you add another entry to the middle of the list

`list = ["apples","bananas", "grapes", "peaches"]`

then you apply by the 2nd time and suddenly, Terraform is (2 create, 0 changes, 1 destroy). The "peaches" will be destroyed and created again.